librarian/data/meta/archive.py

```diff
@@ -0,0 +1,673 @@
+"""
+Public API for querying and modifying file metadata.
+
+Copyright 2014-2015, Outernet Inc.
+Some rights reserved.
+
+This software is free software licensed under the terms of GPLv3. See COPYING
+file that comes with the source code, or http://www.gnu.org/licenses/gpl.txt.
+"""
+import functools
+import logging
+import os
+
+from ...core.exts import ext_container as exts
+from ...core.utils import batched, as_iterable
+from .contenttypes import ContentTypes
+from .processors import Processor, NO_LANGUAGE, DIRECTORY_TYPE, FILE_TYPE
+from .utils import ancestors_of
+from .wrapper import MetaWrapper
+
+
+class FSWriter(object):
+    """
+    Create or update a file system object while guaranteeing that all
```

Comment:
The docstring should reads as if the ancestors will be created on the disk. It should be tweaked to clarify that entries on database will be created